data_source_ssh_tunnel.go

```diff
@@ -26,10 +27,17 @@ func dataSourceSSHTunnel() *schema.Resource {
 				Required:    true,
 				Description: "The hostname",
 			},
+			"ssh_agent": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "use ssh agent",
+				Default:     false,
+			},
 			"private_key": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
```

Comment:
I don't think that the ConflictsWith flag is needed here since we evaluate SSH_AUTH_SOCK and the key itself later, but we definitely should check if at least one auth method is set. 
I see the logic like this:
- Always try ssh-agent unless 'ssh_agent = false' set in TF
- If 'private_key' set use private_key first
- Check if at least one is set.

Reply:
You are right. I wrote this comment before I wrote the other comments, and forgot to change it. :)